omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/ConceptReferenceTermSearchHandler1_9.java

```diff
@@ -62,41 +69,101 @@ public PageableResult search(RequestContext context) throws ResponseException {
 		String source = context.getParameter("source");
 		String code = context.getParameter("code");
 		String name = context.getParameter("name");
-		
+		String searchType = context.getParameter("searchType");
+
+		if(searchType!=null&&!SEARCH_TYPE_EQUALS.equals(searchType)&&!SEARCH_TYPE_STARTS_WITH.equals(searchType)){

```

Comment:
I thought the ticket says support searching by source and not doing partial searches